ddtrace/internal/flare.py

```diff
@@ -0,0 +1,162 @@
+    else:
+        # There was an exception
+        agent_error = "Exception raised: %s" % str(resp)
+
+    is_venv = in_venv()
+
+    packages_available = {p.project_name: p.version for p in pkg_resources.working_set}
+    integration_configs = {}
+    for module in monkey.PATCH_MODULES:
+        # TODO: this check doesn't work in all cases... we need a mapping
+        #       between the module and the library name.
+        module_available = module in packages_available
+        enabled = monkey.PATCH_MODULES[module]
+        module_instrumented = module in monkey._PATCHED_MODULES
+        module_imported = module in sys.modules
+
+        if enabled:
+            # DEV: integration configs aren't added until the integration module is
+            #      imported.
+            config = ddtrace.config._config.get(module, "N/A")
+        else:
+            config = None
+
+        if module_available:
+            integration_configs[module] = dict(
+                enabled=enabled,
+                instrumented=module_instrumented,
+                module_available=module_available,
+                module_version=packages_available[module],
+                module_imported=module_imported,
+                config=config,
+            )
+        else:
+            # Use N/A here to avoid the additional clutter of an entire
+            # config dictionary for a module that isn't available.
+            integration_configs[module] = "N/A"
+
+    pip_version = packages_available.get("pip", "pip not available")
+
+    return dict(
+        # Timestamp UTC ISO 8601
+        date=str(datetime.datetime.utcnow().isoformat()),
+        # eg. "Linux", "Darwin"
+        os_name=platform.system(),
+        # eg. 12.5.0
+        os_version=platform.release(),
+        is_64_bit=sys.maxsize > 2 ** 32,
+        architecture=platform.architecture()[0],
+        vm=platform.python_implementation(),
+        version=ddtrace.__version__,
+        lang="python",
+        lang_version=platform.python_version(),
+        pip_version=pip_version,
+        in_virtual_env=is_venv,
+        agent_url=agent_url,
+        agent_error=agent_error,
+        env=ddtrace.config.env or "",
+        enabled=formats.asbool(os.getenv("DATADOG_TRACE_ENABLED") or "true"),
+        service=ddtrace.config.service or "",
+        debug=formats.asbool(os.getenv("DATADOG_TRACE_DEBUG")),
+        enabled_cli="ddtrace" in os.getenv("PYTHONPATH", ""),
+        analytics_enabled=ddtrace.config.analytics_enabled,
+        log_injection_enabled=ddtrace.config.logs_injection,
+        health_metrics_enabled=ddtrace.config.health_metrics_enabled,
+        dd_version=ddtrace.config.version or "",
+        priority_sampling_enabled=tracer.priority_sampler is not None,
+        global_tags=os.getenv("DD_TAGS", ""),
+        tracer_tags=tags_to_str(tracer.tags),
+        integrations=integration_configs,
+        profiling_enabled=True,
```

Comment:
@jd it'd be great to have profiling information as well. Do you have ideas for:

1) What we can log (we can probably do most of the environment variables)

2) Point(s) where it makes sense to log (maybe in `Profiler()` post-init?)

Reply:
Env vars would be good.
I'd love to have gevent status too if that's possible because it really is a different beast when it comes to profiling.